sklearn/_callbacks.py

```diff
@@ -0,0 +1,58 @@
+# License: BSD 3 clause
+from typing import List, Callable
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+CALLBACK_PARAM_TYPES = {
+    "n_iter": int,
+    "max_iter": int,
+    "loss": (float, dict),
+    "score": (float, dict),
+    "validation_loss": (float, dict),
+    "validation_score": (float, dict),
+    "coef": np.ndarray,
+    "intercept": (np.ndarray, float)
+}
+
+
+def _check_callback_params(**kwargs):
+    invalid_params = []
+    invalid_types = []
+    for key, val in kwargs.items():
+        if key not in CALLBACK_PARAM_TYPES:
+            invalid_params.append(key)
+        else:
+            val_types = CALLBACK_PARAM_TYPES[key]
+            if not isinstance(val, val_types):
+                invalid_types.append(
+                    f"{key}={val} is not of type {val_types}"
+                )
+    msg = ""
+    if invalid_params:
+        msg += ("Invalid callback parameters: {}, must be one of {}. ").format(
+                ", ".join(invalid_params),
+                ", ".join(CALLBACK_PARAM_TYPES.keys())
+        )
+    if invalid_types:
+        msg += "Invalid callback parameters: " + ", ".join(invalid_types)
+    if msg:
+        raise ValueError(msg)
+
+
+def _eval_callbacks(callbacks: List[Callable], **kwargs) -> None:
+    if callbacks is None:
+        return
```

Comment:
Shouldn't we rely on `callbacks` being an empty list instead of subcasing with None?

Or maybe you are anticipating a future where `callbacks=None` would be a default argument to estimators?

Reply:
Yes, we switched to the case when `callbacks=None` when missing.